File: src/Application_Mobile/model/Question.java
package model;

/**
 * Cette classe permet de représenter une Question à ne pas oublier.
 * 
 * @author Christophe
 * @version 1.0
 */
public class Question {

	/*
	 * Le numéro d'identification de la question.
	 */
	private int idQuestion;

	/**
	 * Le nom de la question.
	 */
	private String nomQuestion;

	/**
	 * Initialise les données de la nouvelle question créée.
	 * 
	 * @param id  le numéro d'identification de la question
	 * @param nom le nom de la question
	 */
	public Question(int id, String nom) {
		this.setIdQuestion(id);
		this.setNomQuestion(nom);
	}

	/**
	 * Permet d'accéder au numéro d'identification de la question.
	 * 
	 * @return le numéro d'identification de la question
	 */
	public int getIdQuestion() {
		return this.idQuestion;
	}

	/**
	 * Renseigne le numéro d'identification de la question.
	 * 
	 * @param id le numéro d'identification de la question
	 */
	public void setIdQuestion(int id) {
		this.idQuestion = id;
	}

	/**
	 * Permet d'accéder au nom de la questionk.
	 * 
	 * @return le nom de la question
	 */
	public String getNomQuestion() {
		return this.nomQuestion;
	}

	/**
	 * Renseigne le nom de la question.
	 * 
	 * @param nom le nom de la question
	 */
	public void setNomQuestion(String nom) {
		this.nomQuestion = nom;
	}

	/**
	 * Représente textuellement les données de la question.
	 */
	public String toString() {
		String question = "QUESTION : " + this.idQuestion;
		String nom = "\n\tNom : " + this.nomQuestion;

		return question + nom;
	}

}

File: src/Site_Web/rosetta_site_web/src/api/objets/Question.php
<?php

/** 
 * Question est une classe de représentant la table Question de la base de données.
 * 
 * Question est une classe permettant de renseigner et obtenir les informations
 * d'une question (un élément de la Table).
 * 
 * @author Garcia Christophe
 * @version $Revision: 1.2 $
 * @access public
 */
class Question {
    
    /**
     * Identifiant de la question
     * 
     * @var int
     * @access private
     */
    private $idQuestion;

    /**
     * Nom de la question
     * 
     * @var string
     * @access private
     */
    private $nomQuestion;

    /**
     * Méthode magique __construct()
     * 
     * Permet de créer une nouvelle question
     */
	public function __construct () {
    }

    /**
     * Méthode d'initialisation init()
     * 
     * Iniatilise les données de la question
     * 
     * @param int $id
     * @param string $nom
     */
	public function init ($id, $nom) {
        $this->__set('id', $id);
        $this->__set('nom', $nom);
    }

    /**
     * Méthode magique __get()
     * 
     * Retourne la valeur de la propriété de la question
     * 
     * @param string $propriete
     * @return int|string
     * @throws Exception
     */
    public function __get($propriete) {
        if ($propriete === 'id') {
            return $this->idQuestion;
        } else if ($propriete === 'nom') {
            return $this->nomQuestion;
        } else {
            throw new Exception('Question: __get(int|string): propriété invalide.');
        }
    }

    /**
     * Méthode magique __set()
     * 
     * Fixe la valeur de la propriété appelée à la question
     * 
     * @param string $propriete
     * @param mixed $valeur
     * @throws Exception
     */
    public function __set($propriete, $valeur) {
        if ($propriete === 'id') {
            $this->idQuestion = $valeur;
        } else if ($propriete === 'nom' && is_string($valeur)) {
            $this->nomQuestion = $valeur;
        } else {
            throw new Exception('Question: __set(string, mixed): propriété ou valeur invalide.');
        }
    }
    
    /**
     * Méthode magique __toString()
     * 
     * Retourne le nom de la question
     * 
     * @return string
     */
	public function __toString() {
        return "Nom : " . $this->nomQuestion . "\r\n";
	}

}

?>
